src/Features/Core/Portable/ChangeAccessibilityModifier/AbstractChangeAccessibilityModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,116 @@
+            {
+                return;
+            }
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var declaredSymbol = semanticModel.GetDeclaredSymbol(diagnosticNode, cancellationToken);
+            if (declaredSymbol is not (IPropertySymbol or IMethodSymbol or IEventSymbol))
+            {
+                return;
+            }
+
+            var diagnostic = context.Diagnostics[0];
+
+            if (declaredSymbol.IsOverride)
+            {
+                // only show accessibility of base definition for override
+                var original = declaredSymbol.GetOverriddenMember();
+                if (original is null)
+                {
+                    return;
+                }
+
+                var accessibility = original.ComputeResultantAccessibility(declaredSymbol.ContainingType);
+
+                if (accessibility == Accessibility.Internal
+                    && !original.ContainingAssembly.GivesAccessTo(declaredSymbol.ContainingAssembly))
+                {
+                    // not able to override inaccessible member - return
+                    return;
+                }
+
+                if (accessibility == Accessibility.Private)
+                {
+                    // should be unreachable - return for robustness
+                    return;
+                }
+
+                context.RegisterCodeFix(CreateAction(accessibility), diagnostic);
+                return;
+            }
+
+            context.RegisterCodeFix(
+                new MyNestedAction(
+                    _title,
+                    ImmutableArray.Create(
+                        CreateAction(Accessibility.Public),
+                        CreateAction(Accessibility.Protected),
+                        CreateAction(Accessibility.Internal),
+                        CreateAction(Accessibility.ProtectedOrInternal),
+                        CreateAction(Accessibility.ProtectedAndInternal)),
+                    isInlinable: true),
+                diagnostic);
+
+            CodeAction CreateAction(Accessibility accessibility)
+                => new MyCodeAction(
+                    string.Format(_titleFormat, GetText(accessibility)),
+                    async ct =>
+                    {
+                        var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+                        editor.SetAccessibility(diagnosticNode, accessibility);
+                        return editor.GetChangedDocument();
+                    });
+        }
+
+        protected abstract string GetText(Accessibility accessibility);
+
+        private class MyNestedAction : CodeAction.CodeActionWithNestedActions
+        {
+            public MyNestedAction(string title, ImmutableArray<CodeAction> nestedActions, bool isInlinable)
+                : base(title, nestedActions, isInlinable, CodeActionPriority.High)
```

Comment:
being high pri worries me.  why do we need this?  if you're fixing an error, this should naturally appear near hte top.

For context: only *two* fixes in all of roslyn are high pri, and that's because they're so ridiculously important, we cannot allow anything to supersede them for muscle memory reasons.